docs/vivisect/devguides/gettingstarted.rst

```diff
@@ -36,14 +66,287 @@ If you've already read a file into memory, you can instead use the `loadFromFd`
+
+    In [17]: vw.config.viv.parsers.ihex
+    Out[17]: <envi.config.EnviConfig at 0x7fe9d2760070>
+
+    In [18]: print(vw.config.viv.parsers.ihex.reprConfigPaths())
+    Valid Config Entries:
+        .arch = cc8051
+        .offset = 0
+        .bigend = False
+
+    Valid Config Paths:
+    
+
+    In [19]: vw.config.viv.parsers.ihex.arch='arm'
+
+
+Once you have configured the necessary items, load your file::
+
+    In [31]: vw.loadFromFile('/home/atlas/work/firmware.hex')                                      
+
+    In [32]: vw.getMeta('Architecture')                                                                                                
+    Out[32]: 'arm'
+
+    In [32]: vw.analyze()
+
+When you're happy with your workspace, be sure to save it::
+
+    In [33]: vw.saveWorkspace(fullsave=True)
+
+To save a workspace, `vw.saveWorkspace()` is used.  The "fullsave=True" means to write a complete file, instead of saving incrementally.  For the first time save, this is important, as it places the header on the workspace file which tells Viv what kind of file it is.
+
+No filename is given, but the filename to be written is stored in the workspace metadata.  The default name is the last file loaded into the workspace + ".viv".  You can see and modify this filename like so::
+
+    In [36]: vw.getMeta('StorageName')                                                                                                 
+    Out[36]: '/home/atlas/work/firmware.hex.viv'
+
+    In [37]: vw.setMeta('StorageName', '/home/atlas/work/firmware.hex-clean-211205.viv')
+
+    In [38]: vw.saveWorkspace(fullsave=True)                                                                                           
+
+
+Loading an ELF and Working With Functions
+=========================================
+
+Getting started working with files is really quite easy.  Cherry-picking from the illustrations above::
+
+    import vivisect
+    vw = vivisect.VivWorkspace()
+    vw.loadFromFile('/bin/chown')
+    vw.analyze()
+    vw.setMeta('StorageName', '/home/atlas/work/chown-new.viv')
+    vw.saveWorkspace(True)
+
+or:
+    In [62]: vw = vivisect.VivWorkspace()                                                                                              
+
+    In [63]: vw.loadFromFile('/bin/chown')                                                                                             
+    Out[63]: 'chown'
+
+    In [64]: vw.getMeta('StorageName')                                                                                                 
+    Out[64]: '/bin/chown.viv'
+
+    In [65]: vw.analyze() 
+
+    In [66]: vw.setMeta('StorageName', '/home/atlas/work/chown-new.viv') 
+
+    In [67]: vw.saveWorkspace(True) 
+
+`VivWorkspace.getFunctions()` returns a list of Virtual Addresses (va's) for the beginning of each function::
+
+    In [67]: vw.getFunctions()                                                                                                         
```

Comment:
As a formatting note, it's a little jarring and messy to mix raw python with the ipython markups. I'd like to standardize on one. I'm not particularly picky about which, but I'd like to be consistent.

Reply:
i understand the desire for consistency.  i'm not sure i agree that they should be treated the same, especially in this format (markdown).  
ipython gives context and a real feeling like you're "in the terminal with the author" whereas the others are easily copy/pasted.

i can be convinced... but i'm not certain that i am yet.  

thoughts?